src/index.js

```diff
@@ -0,0 +1,748 @@
+   * @param {Object} params={} - An object containing the id of the desired metaproperty and parameters accepted by the
+   * API to narrow the query for options
+   * @param {String} params.id - The id of the desired metaproperty.
+   * @return {Promise<array>} Response - Returns a Promise that, when fulfilled, will either return an Array with the
+   * metaproperties or an Error with the problem.
+   */
+  getMetapropertyOptions({ id, ...params } = {}) {
+    if (!id) {
+      return rejectValidation('metapropertyOption', 'id');
+    }
+
+    return this.api.send('GET', `api/v4/metaproperties/${id}/options/`, params);
+  }
+
+  /**
+   * Get the download url of original file.
+   * @see {@link https://bynder.docs.apiary.io/#reference/download/download-operations/retrieve-asset-download-location|API Call}
+   * @param {Object} params - An object containing the parameters accepted by the API to change in the asset.
+   * @param {String} params.id - The id of the desired asset.
+   * @returns {Promise}
+   */
+  getMediaDownloadUrl({ id, ...params } = {}) {
+    if (!id) {
+      return rejectValidation('media', 'id');
+    }
+
+    return this.api.send('GET', `api/v4/media/${id}/download`, { ...params });
+  }
+
+  /**
+   * Finalises the file upload when all chunks finished uploading and registers it in Bynder.
+   * @see {@link https://bynder.docs.apiary.io/#reference/upload-assets/4-finalise-a-completely-uploaded-file/finalise-a-completely-uploaded-file}
+   * @param {String} fileId Unique file identifier
+   * @param {String} fileName Original file name
+   * @param {Number} chunksCount Number of chunks
+   * @param {Number} fileSize File byte size
+   * @return {Promise<object>} Object containing the correlation ID (`correlationId`) and file ID (`fileId`) of the upload
+   */
+  finaliseUpload(fileId, fileName, chunksCount, fileSize) {
+    return this.api.send('POST', `v7/file_cmds/upload/${fileId}/finalise_api`, {
+      chunksCount, fileName, fileSize,
+      intent: DEFAULT_UPLOAD_INTENT,
+      sha256: this._sha256
+    })
+      .then(response => response.headers['x-api-correlation-id']);
+  }
+
+  /**
+   * Prepares the remote env to upload a new file
+   * @return {Promise<string>} Unique file identificator
+   */
+  prepareUpload() {
+    return this.api.send('POST', 'v7/file_cmds/upload/prepare')
+      .then(response => response.file_id);
+  }
+
+  /**
+   * Saves a media asset in Bynder. If media id is specified in the data a new version of the asset will be saved.
+   * Otherwise a new asset will be saved.
+   * @see {@link https://bynder.docs.apiary.io/#reference/upload-assets/4-finalise-a-completely-uploaded-file/save-as-a-new-asset}
+   * @param {Object} data Asset data
+   * @param {String} data.brandId Brand ID
+   * @param {String} data.fileId File ID
+   * @param {String} data.mediaId Media ID
```

Comment:
Passing a `name` in the data can be useful as well, since otherwise the asset will have no name.

From the api doc: "Warning: The asset will have no name if this is left empty."
https://bynder.docs.apiary.io/#reference/upload-assets/6-saving-a-completed-upload/save-as-a-new-asset

Reply:
What do you think of this? In some other SDKs we just pass the filename as the asset name for example.

Reply:
We could align it with the other SDKs, but that would mean more breaking changes for this new version. Either way I'm OK with any, or even both.

Reply:
Name can be sent as well :)

Reply:
Yeah but I meant it's not included in the `@param {String} ...` docstrings. So I thought that could be helpful for the user to see what relevant fields there are to set.

Reply:
Is a couple of lines [below](https://github.com/Bynder/bynder-js-sdk/blob/add/file-service/src/index.js#L639).